### PR TITLE
Feat/room storage

### PR DESCRIPTION
**Pull request by mailo001**

### CODE REVIEW

File: server/pkg/postgres/room_storage.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	if err != nil {
+		return nil, nil, fmt.Errorf("postgres: could not intitialize postgres pool: %w", err)
+	}
+	storage := &RoomStorage{
+		connPool: pool,
+	}
+	return storage, cleanup, nil
+}
+
+// CreateRoom implements business logic
```
Comment (Gloomwidow) [inline server/pkg/postgres/room_storage.go:34]: What business logic? At least "for creating room" should be enough.

File: server/pkg/postgres/room_storage.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		room.RoomNumber,
+		hotelID,
+	).Scan(&id)
+	if err != nil {
+		return 0, fmt.Errorf("postgres: could not insert offer: %w", err)
+	}
+	return id, nil
+}
+
+// DeleteRoom implements business logic
```
Comment (Gloomwidow) [inline server/pkg/postgres/room_storage.go:55]: Same as above.

File: server/pkg/postgres/room_storage.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		roomID,
+	)
+	if err == pgx.ErrNoRows {
+		return bookly.RoomNotFound, nil
+	}
+
+	return bookly.RoomSuccess, nil
+}
+
+// GetAllHotelRoom implements business logic
```
Comment (Gloomwidow) [inline server/pkg/postgres/room_storage.go:94]: Again, refer to comment above.

File: server/pkg/postgres/room_storage.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+func (r RoomStorage) GetAllHotelRoom(ctx context.Context, hotelID int) ([]*bookly.Room, error) {
```
Comment (Gloomwidow) [inline server/pkg/postgres/room_storage.go:95]: I think `GetAllHotelRooms` should be a better name for that.

File: server/pkg/postgres/room_storage.go (hunk 1 above)
Comment (mailo001) [reply server/pkg/postgres/room_storage.go:34]: Good point

File: server/pkg/postgres/room_storage.go (hunk 2 above)
Comment (mailo001) [reply server/pkg/postgres/room_storage.go:55]: Good point

File: server/pkg/postgres/room_storage.go (hunk 3 above)
Comment (mailo001) [reply server/pkg/postgres/room_storage.go:94]: Good point

File: server/pkg/postgres/room_storage.go (hunk 4 above)
Comment (mailo001) [reply server/pkg/postgres/room_storage.go:95]: I agree

Comment (mailo001) [reply server/pkg/postgres/room_storage.go:95]: I agree

**Pull request merged**